### PR TITLE
[REVIEW] Fix `dask_cudf` categorical nonempty meta handling

### LINKED ISSUE #4608: [BUG] dask-cudf groupby yields error on category column
**Describe the bug**
@taureandyernv @randerzander for viz.

I am trying to groupby with dask-cudf on a category column. It yields the error below. 
Dask version is 2.12.0

**Steps/Code to reproduce bug**
```
import dask
import cudf
import dask_cudf as dcu
df = dask.datasets.timeseries()
ddf = dcu.from_dask_dataframe(df)
ddf.dtypes
id            int64
name     object
x            float64
y            float64
dtype: object

ddf['name']=ddf['name'].astype('category')

ddf.dtypes
id            int64
name    category
x           float64
y           float64

ddf.groupby('name').agg({'x': ['mean', 'max'], 'y': ['mean', 'count']}).compute()
```
We get the following error:

```
......
......
IndexError: 

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-25-53cb03531f6d> in <module>
----> 1 ddf.groupby('name').agg({'x': ['mean', 'max'], 'y': ['mean', 'count']}).compute()

~/yes/envs/rapids_13_tpcx/lib/python3.7/site-packages/dask/dataframe/groupby.py in agg(self, arg, split_every, split_out)
   1756     @derived_from(pd.core.groupby.DataFrameGroupBy)
   1757     def agg(self, arg, split_every=None, split_out=1):
-> 1758         return self.aggregate(arg, split_every=split_every, split_out=split_out)
   1759 
   1760 

~/yes/envs/rapids_13_tpcx/lib/python3.7/site-packages/dask/dataframe/groupby.py in aggregate(self, arg, split_every, split_out)
   1751 
   1752         return super(DataFrameGroupBy, self).aggregate(
-> 1753             arg, split_every=split_every, split_out=split_out
   1754         )
   1755 

~/yes/envs/rapids_13_tpcx/lib/python3.7/site-packages/dask/dataframe/groupby.py in aggregate(self, arg, split_every, split_out)
   1532             split_out=split_out,
   1533             split_out_setup=split_out_on_index,
-> 1534             sort=self.sort,
   1535         )
   1536 

~/yes/envs/rapids_13_tpcx/lib/python3.7/site-packages/dask/dataframe/core.py in apply_concat_apply(args, chunk, aggregate, combine, meta, token, chunk_kwargs, aggregate_kwargs, combine_kwargs, split_every, split_out, split_out_setup, split_out_setup_kwargs, sort, **kwargs)
   4891 
   4892     if meta is no_default:
-> 4893         meta_chunk = _emulate(chunk, *args, udf=True, **chunk_kwargs)
   4894         meta = _emulate(aggregate, _concat([meta_chunk]), udf=True, **aggregate_kwargs)
   4895     meta = make_meta(

~/yes/envs/rapids_13_tpcx/lib/python3.7/site-packages/dask/dataframe/core.py in _emulate(func, *args, **kwargs)
   4936     """
   4937     with raise_on_meta_error(funcname(func), udf=kwargs.pop("udf", False)):
-> 4938         return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
   4939 
   4940 

~/yes/envs/rapids_13_tpcx/lib/python3.7/contextlib.py in __exit__(self, type, value, traceback)
    128                 value = type()
    129             try:
--> 130                 self.gen.throw(type, value, traceback)
    131             except StopIteration as exc:
    132                 # Suppress StopIteration *unless* it's the same exception that

~/yes/envs/rapids_13_tpcx/lib/python3.7/site-packages/dask/dataframe/utils.py in raise_on_meta_error(funcname, udf)
    191         )
    192         msg = msg.format(" in `{0}`".format(funcname) if funcname else "", repr(e), tb)
--> 193         raise ValueError(msg)
    194 
    195 

ValueError: Metadata inference failed in `_groupby_apply_funcs`.

You have supplied a custom function and Dask is unable to 
determine the type of output that that function returns. 

To resolve this please provide a meta= keyword.
The docstring of the Dask function you ran should have more information.

Original error is below:
------------------------
IndexError()
.....
```
I tried to add `meta=keyword` in the `agg()` but it did not work.  

**Expected behavior**
```
df.groupby('name').agg({'x': ['mean', 'max'], 'y': ['mean', 'count']}).compute().head()
```
image

**Environment overview (please complete the following information)**
 - Environment location:  Conda
 - Method of cuDF install: Conda - rapids nightly (cudf 0.13.0a200317)

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

### PR DESCRIPTION
**Pull request by shwina**
Fixes #4608

### CODE REVIEW

File: python/dask_cudf/dask_cudf/backends.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _nonempty_series(s, idx=None):
         idx = _nonempty_index(s.index)
     dtype = s.dtype
     if is_categorical_dtype(dtype):
-        categories = s._column.categories
+        categories = ["a", "b"]
```
Comment (shwina) [inline python/dask_cudf/dask_cudf/backends.py]: @rjzamora would appreciate your input on whether this is OK to do

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -658,7 +658,7 @@ def _copy_categories(self, other, include_index=True):
             ):
                 self._data[name] = build_categorical_column(
                     categories=other_col.categories,
-                    codes=col,
+                    codes=as_column(col.base_data, dtype=col.dtype),
```
Comment (kkraus14) [inline python/cudf/cudf/core/frame.py]: This feels super janky.

Comment (shwina) [reply python/cudf/cudf/core/frame.py]: What about it other than the verbosity? The alternative is to make this more expensive and use `col.data` and `col.mask` instead :(

Comment (kkraus14) [reply python/cudf/cudf/core/frame.py]: I think moreso that there's ambiguity of how offsets / size are handled for codes vs the CategoricalColumn itself. I.E. imagine wanting to construct a CategoricalColumn with an offset from a codes NumericalColumn that also has a different offset. That isn't possible with this.

Comment (shwina) [reply python/cudf/cudf/core/frame.py]: Removed this entirely from the PR

**Pull request merged**